common/src/Microsoft.Azure.IIoT.Agent.Framework/src/Jobs/Hub/IoTHubJobConfigurationHandler.cs

```diff
@@ -46,6 +46,39 @@ public class IoTHubJobConfigurationHandler : IJobEventHandler {
                     };
                     await _ioTHubTwinServices.CreateOrUpdateAsync(deviceTwin, true);
                 }
+            }
+            catch (Exception ex) {
+                _logger.Error(ex, "Error while creating IoT Device.");
+            }
+        }
+
+        /// <inheritdoc/>
+        public async Task OnJobScopingAsync(IJobService manager, JobInfoModel job, string workerId) {
+            if (job.JobConfiguration?.IsObject != true) {
+                return;
+            }
+            try {
+                var edgeDeviceTwin = await _ioTHubTwinServices.FindAsync(workerId.Split("_publisher")[0]);
```

Comment:
>workerId.Split("_publisher")[0] [](start = 73, length = 31)

Avoid potential crash, do not access index zero

Reply:
right, I added an argument validation to make sure the workerId is not null. The split itself will always return an array with size at lease 1, therefore index zero always exist.